dsgrid/query/models.py

```diff
@@ -0,0 +1,396 @@
+import abc
+import enum
+from pathlib import Path
+from typing import Any, List, Optional, Set, Union
+
+import pyspark.sql.functions as F
+from pydantic import Field, root_validator, validator
+from semver import VersionInfo
+
+from dsgrid.data_models import DSGBaseModel
+
+from dsgrid.dimension.base_models import DimensionType
+from dsgrid.dimension.dimension_filters import make_dimension_filter, DimensionFilterBaseModel
+from dsgrid.utils.files import compute_hash, load_data
+
+
+class FilteredDatasetModel(DSGBaseModel):
+
+    dataset_id: str = Field(description="Dataset ID")
+    filters: List[Any] = Field(
+        description="Dimension filters to apply to the dataset'",
+    )
+
+
+# TODO: setting alias on a groupBy column doesn't work on a per-dataset aggregation
```

Comment:
remove alias option from per-dataset agg model then?

Reply:
We need it. But, I will add code to fail an operation that includes alias for per-dataset agg.